master/buildbot/status/build.py

```diff
@@ -267,6 +271,27 @@ def getBuildUrl(self):
         return self.master.status.getURLForThing(self)['path'] if 'path' in self.master.status.getURLForThing(
             self) else ''
 
+    def getTopBuildUrl(self):
+        d = self.master.db.buildrequests.getTopBuildData(self.buildChainID)
+
+        def createTopBuildUrl(build_chain):
+            builder_name = build_chain['buildername']
+            build_number = build_chain['build_number']
+
+            if self.number == build_number:
```

Comment:
I think you need to add here the buildername as well, as the link are not created if 2 different builders has the same number.